File: csharp_new/src/Vista.SDK/VisVersion.cs
namespace Vista.SDK;

public static class VisVersionExtensions
{
    private static readonly uniffi.bindings.VisVersionExtensions _extensions = new();

    public static string ToVersionString(this VisVersion version) => _extensions.ToVersionString(version);
}

public static class VisVersions
{
    private static readonly uniffi.bindings.VisVersions _visVersions = new();

    public static VisVersion Parse(string input) => _visVersions.Parse(input);

    public static VisVersion[] All => _visVersions.All().ToArray();
}

File: rust/bindgen/src/vis_version.rs
use std::str::FromStr;

use vista_sdk::vis::VisVersion;

use crate::{VisVersionExtensions, VisVersions};

impl VisVersionExtensions {
    pub fn new() -> Self {
        VisVersionExtensions {}
    }

    pub fn to_version_string(&self, version: VisVersion) -> String {
        format!("{}", version)
    }
}

impl VisVersions {
    pub fn new() -> Self {
        VisVersions {}
    }

    pub fn parse(&self, input: &str) -> VisVersion {
        VisVersion::from_str(input).unwrap()
    }
    pub fn all(&self) -> Vec<VisVersion> {
        VisVersion::ALL.to_vec()
    }
}
